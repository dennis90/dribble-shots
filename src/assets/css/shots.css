.shots {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
}

.shot {
  width: 10rem;
  height: 7.5rem;
  display: block;
  margin-bottom: 1.5rem;
  background-size: 100%;
  cursor: pointer;
}

@media(min-width: 600px) {
  .shot {
    width: 16.25rem;
    height: 12.1875rem;
  }
}

.shot-footer {
  display: flex;
  position: relative;
  justify-content: flex-end;
  top: 75%;
  height: 25%;
  background-color: rgba(0, 0, 0, 0.3);
}

.shot-views {
  margin: auto 0;
  margin-right: 1rem;
  color: white;
  font-weight: bold;
  background: url('../img/eye.png') no-repeat;
  background-position: left;
  padding-left: 2rem;
}
