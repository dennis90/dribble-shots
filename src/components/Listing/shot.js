import React from 'react'

import ShotFooter from './shot-footer'


const Shot = ({shot}) => {
  const getStyle = () => ({
      backgroundImage: `url(${shot.images.normal})`
  })

  return (
    <div className="shot" style={getStyle()}>
      <ShotFooter shot={shot} />
    </div>
  )
}

export default Shot
