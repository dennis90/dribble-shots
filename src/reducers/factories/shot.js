import {Factory} from 'rosie'

export const ShotFactory = new Factory()
  .sequence('id')
  .sequence('title', index => `Shot #${index}`)
  .sequence('description', index => `Shot Description #${index}`)
  .attrs({
    images: {
      hidpi: 'https://dribbble/images/hidpi',
      normal: 'https://dribbble/images/normal',
    },
    views_count: 100,
    likes_count: 100,
  })
  .attr('user', ['user'], user => UserFactory.build(user))

export const UserFactory = new Factory()
  .sequence('name', index => `User #${index}`)
  .attrs({
    avatar_url: 'https://dribbble.com/user/avatar.png',
    html_url: 'https://dribbble.com/user/',
  })
