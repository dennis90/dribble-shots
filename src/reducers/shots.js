import types from '../actionTypes'
import {OrderedMap} from 'immutable'

export default function shots(state=OrderedMap(), action) {
  switch (action.type) {
    case types.SHOTS_FETCHED:
      return action.payload.reduce(
        (nextState, shot) => nextState.set(shot.id, shot), state
      )

    default:
      return state
  }
}
