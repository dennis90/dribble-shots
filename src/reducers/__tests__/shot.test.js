import shots from '../shots'
import types from '../../actionTypes/'
import {ShotFactory} from '../factories/shot'

describe('Shots Reducer', () => {
  it('should handle initial state', () => {
    const foundState = shots(undefined, {type: undefined})
    expect(foundState).toMatchSnapshot()
  })

  it('should store received shots at `SHOTS_FETCHED` action type', () => {
    const action = {type: types.SHOTS_FETCHED, payload: ShotFactory.buildList(3)}
    const foundState = shots(undefined, action)
    expect(foundState).toMatchSnapshot()
  })
})
