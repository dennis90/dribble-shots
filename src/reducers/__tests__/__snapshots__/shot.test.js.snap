// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shots Reducer should handle initial state 1`] = `
Immutable.OrderedMap {
}
`;

exports[`Shots Reducer should store received shots at \`SHOTS_FETCHED\` action type 1`] = `
Immutable.OrderedMap {
  1: Object {
    "description": "Shot Description #1",
    "id": 1,
    "images": Object {
      "hidpi": "https://dribbble/images/hidpi",
      "normal": "https://dribbble/images/normal",
    },
    "likes_count": 100,
    "title": "Shot #1",
    "user": Object {
      "avatar_url": "https://dribbble.com/user/avatar.png",
      "html_url": "https://dribbble.com/user/",
      "name": "User #1",
    },
    "views_count": 100,
  },
  2: Object {
    "description": "Shot Description #2",
    "id": 2,
    "images": Object {
      "hidpi": "https://dribbble/images/hidpi",
      "normal": "https://dribbble/images/normal",
    },
    "likes_count": 100,
    "title": "Shot #2",
    "user": Object {
      "avatar_url": "https://dribbble.com/user/avatar.png",
      "html_url": "https://dribbble.com/user/",
      "name": "User #2",
    },
    "views_count": 100,
  },
  3: Object {
    "description": "Shot Description #3",
    "id": 3,
    "images": Object {
      "hidpi": "https://dribbble/images/hidpi",
      "normal": "https://dribbble/images/normal",
    },
    "likes_count": 100,
    "title": "Shot #3",
    "user": Object {
      "avatar_url": "https://dribbble.com/user/avatar.png",
      "html_url": "https://dribbble.com/user/",
      "name": "User #3",
    },
    "views_count": 100,
  },
}
`;
